### PR TITLE
updated mapping doc

### PR DESCRIPTION
**Pull request by nickeubank**
Working on improving mapping doc. Part of #281

Anyone have advice on layering maps? Never done myself.

### CODE REVIEW

**Comment by nickeubank** [comment]
@kjordahl @jorisvandenbossche Figuring out mapping by trial and error. Not sure how to overlay layers / if what I say at top of this new mapping page is quite right about what options you can pass. Comments?

**Comment by nickeubank** [comment]
Ah, figured out part of the problem -- map is documented, just not in accessible location. Moving those via #300

File: doc/source/mapping.rst [hunk 1]
```diff
@@ -26,11 +24,63 @@ Chloropleth Maps
...
+Maps with Layers
 -----------------
+
+To overlap different layers, first ensure they share a common CRS (so they will align), then plot.
+
+.. ipython:: python
+    
+    # Look at capitals
+    # Note use of standard `pyplot` line style options 
+    # (points are still "lines" to matplotlib).
```
Comment (jorisvandenbossche) [inline doc/source/mapping.rst]: I don't get this remark?

File: doc/source/mapping.rst [hunk 2]
```diff
@@ -26,11 +24,63 @@ Chloropleth Maps
...
+    # Note use of standard `pyplot` line style options 
+    # (points are still "lines" to matplotlib). 
+    @savefig capitals.png width=5in
+    cities.plot(marker='*', color='green', markersize=5);
+
+    # Check crs
+    cities = cities.to_crs(world.crs)
+
+    # Overlay over country outlines
+    #[anyone know how to do this?]
```
Comment (jorisvandenbossche) [inline doc/source/mapping.rst]: Two possible options:

```
ax =  world.plot()
cities.plot(ax=ax)
```

or

```
fig, ax = plt.subplots()
ax.set_aspect('equal')
world.plot(ax=ax)
cities.plot(ax=ax)
```

I find the second the cleanest, but the drawback is that you then have to know to set the aspect to equal (which should be done for mapping)

**Comment by jorisvandenbossche** [comment]
@nickeubank There is still a problem with the locations of the example data (see https://github.com/geopandas/geopandas/pull/298/commits/142393c60af87c9ba54123d48a178e4cba665332). I didn't get it to work both locally and on readthedocs ..

**Comment by nickeubank** [comment]
@jorisvandenbossche Updated contents.  

Not sure about path -- i've never seen the backend of readthedocs -- is there a place to see output from builds? I'll push with weird path for now.

**Comment by nickeubank** [comment]
@jorisvandenbossche one other thing that'd be nice to add is something about adding basemaps. You ever done that?

**Comment by jorisvandenbossche** [comment]
@nick with basemaps, do you mean the matplotlib toolkit basemap (`mpl_toolkits.basemap`)? Or do you just mean the general idea? (there is eg also the newer cartopy package)

**Comment by nickeubank** [comment]
@jorisvandenbossche Recently I was told on the matplotlib mailing list that `mpl_toolkits.basemap` was largely depreciated and I should use `cartopy`, so that seems like a good choice. I think an example would be nice but I've never worked with it -- just wondering if you have any code snippets.

**Comment by darribas** [comment]
Something like [`smopy`](https://github.com/rossant/smopy) but updated? R has `ggmap` and it works great. It'd be great if `geopandas` had similar functionality (as an option).

**Comment by nickeubank** [comment]
@darribas Something like that would be great. I think `cartopy` is also able to pull in basemaps from various servers. At some point it'd be great if it were fully integrated, but for the moment I'd just love a code snippet from someone who's figured out how to put a basemap under a `geopandas` plot in a way that ensures all the CRS line up and such.

**Comment by jorisvandenbossche** [comment]
Yes, some examples of integrations with other libraries would be great (I also think of eg leaflet), but that can be for other PRs!

@nickeubank WIll merge this one, then you can rebase the other doc PRs (as then the data are included)

**Pull request merged**